taroscript/send.go

```diff
@@ -152,6 +153,107 @@ func PrepareOutputAssets(input *taropsbt.VInput,
 	return nil
 }
 
+// SignVirtualTransaction updates the new asset (the root asset located at the
+// change output in case of a non-interactive or partial amount send or the
+// full asset in case of an interactive full amount send) by creating a
+// signature over the asset transfer, verifying the transfer with the Taro VM,
+// and attaching that signature to the new Asset.
+func SignVirtualTransaction(input *taropsbt.VInput, outputs []*taropsbt.VOutput,
+	signer Signer, validator TxValidator) error {
+
+	prevAssets := commitment.InputSet{
+		input.PrevID: input.Asset,
+	}
+	newAsset := outputs[1].Asset
+	if input.IsSplit {
+		newAsset = outputs[0].Asset
+	}
+
+	// Create a Taro virtual transaction representing the asset transfer.
+	virtualTx, _, err := VirtualTx(newAsset, prevAssets)
+	if err != nil {
+		return err
+	}
+
+	// For each input asset leaf, we need to produce a witness. Update the
+	// input of the virtual TX, generate a witness, and attach it to the
```

Comment:
I think the intention here is to re-sign all the other leaves that live in the same tree as the input?

Reply:
Aaah, I see. Yeah, that makes sense. So we'd actually need to loop over all the assets in the input set. Going to re-add a TODO somewhere for this.